src/test/datascience/kernel-launcher/kernelFinder.vscode.test.ts

```diff
@@ -37,16 +37,31 @@ suite('DataScience - Kernels Finder', () => {
         });
         assert.isUndefined(kernelSpec);
     });
-    test('Python kernel returned if no matching kernel found', async () => {
+    test('Python kernel should not be returned if no matching kernel found (for notebooks)', async () => {
+        const kernelSpec = await kernelFinder.findKernel(resourceToUse, {
```

Comment:
For ipynb we return a python kernel only when we know its a python notebook (based on metadta or the like).